src/operation-poller.ts

```diff
@@ -9,6 +9,7 @@ export interface OperationPollerOptions {
   operationResourceName: string;
   backoff?: number;
   masterTimeout?: number;
+  onPoll?: (operation: OperationResult<any>) => any;
```

Comment:
I think the "really right" way to do this in JS would be to implement/extend an event emitter, but it's functionally just this anyway. :) LGTM

Reply:
My 2c: EventEmitter is generally for when you might need to attach many/arbitrary listeners. If you don't have a need to support multiple listeners, you don't really need an EventEmitter.